### DISCUSSION TITLE
Error and failure while installing search node

### DISCUSSION THREAD

**Discussion by tadole** [Security-Onion-Solutions/securityonion#9605] [General]
I am planning to build in a Distributed configuration. 
I have already installed Manager, and when I try to install the search node next, an error occurs during the installation and fails.  
The version of SecurityOnion ISO is 2.3.190. 
Attached is the part of the [sosetup-error.log](https://github.com/Security-Onion-Solutions/securityonion/files/10461704/sosetup-error.log)  where the error occurred. Could you please help me with a solution?

**Comment by cm-ops** [comment]
Have you tried to re-run setup? Would you be able to share the whole` sosetup.log`?

**Comment by tadole** [comment]
I have tried to re-run setup, but get the same error and setup does not work.
I build in an environment where all servers can connect to the Internet.

I upload sosetup.log. Some information is masked.
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/10476083/sosetup.log)

**Comment by cm-ops** [comment]
I see this in the sosetup.log `[ERROR   ] Source file salt://salt/etc/minion.d/mine_functions.conf not found in saltenv 'base'`. Check the manager at this location for the file - `/opt/so/saltstack/default/salt/salt/etc/minion.d/mine_functions.conf`

**Comment by tadole** [comment]
The following file isn't found in the manager.
/opt/so/saltstack/default/salt/salt/etc/minion.d/mine_functions.conf

I reinstalled the manager and checked sosetup.log but the same error is confirmed.
[ERROR   ] Source file salt://salt/etc/minion.d/mine_functions.conf not found in saltenv 'base'
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/10505847/sosetup.log)

Could you let me know if there is any solution?

**Comment by cm-ops** [comment]
You can add the file with `socore:socore` permissions at `/opt/so/saltstack/default/salt/salt/etc/minion.d/` You can get the contents from 
https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/salt/etc/minion.d/mine_functions.conf and try to run setup again.

**Comment by tadole** [comment]
I cannot find the folder `/opt/so/saltstack/default/salt/salt/etc/minion.d/`.  
`socore:socore` authority seems to have been granted.

```
[masked@masked etc]# pwd
/opt/so/saltstack/default/salt/salt/etc
[masked@masked etc]# ls -la
total 4
drwxr-xr-x. 2 socore socore    6 Jan 25 08:24 .
drwxr-xr-x. 7 socore socore 4096 Jan 25 08:24 ..
```

```
[masked@masked salt]# pwd
/opt/so/saltstack/default/salt/salt
[masked@masked salt]# ls -la
total 56
drwxr-xr-x.  7 socore socore 4096 Jan 25 08:24 .
drwxr-xr-x. 56 socore socore 4096 Jan 25 08:24 ..
-rw-r--r--.  1 socore socore  678 Jan 25 08:24 beacons.sls
drwxr-xr-x.  2 socore socore   26 Jan 25 08:24 engines
drwxr-xr-x.  2 socore socore    6 Jan 25 08:24 etc
drwxr-xr-x.  2 socore socore  164 Jan 25 08:24 files
-rw-r--r--.  1 socore socore   50 Jan 25 08:24 helper-packages.sls
-rw-r--r--.  1 socore socore  578 Jan 25 08:24 init.sls
-rw-r--r--.  1 socore socore   89 Jan 25 08:24 lasthighstate.sls
-rw-r--r--.  1 socore socore 2162 Jan 25 08:24 map.jinja
-rw-r--r--.  1 socore socore  299 Jan 25 08:24 master.defaults.yaml
-rw-r--r--.  1 socore socore  866 Jan 25 08:24 master.sls
-rw-r--r--.  1 socore socore  391 Jan 25 08:24 minion-check.sls
-rw-r--r--.  1 socore socore  562 Jan 25 08:24 minion.defaults.yaml
-rw-r--r--.  1 socore socore 3253 Jan 25 08:24 minion.sls
-rw-r--r--.  1 socore socore  103 Jan 25 08:24 minion-state-apply-test.sls
-rw-r--r--.  1 socore socore 2642 Jan 25 08:24 python3-influxdb.sls
drwxr-xr-x.  2 socore socore   31 Jan 25 08:24 scripts
drwxr-xr-x.  2 socore socore   36 Jan 25 08:24 service
```

**Reply by cm-ops** [reply]
You can create the directory as well, then add the file.

**Reply by tadole** [reply]
After adding the file and reinstalling, the search node installation completed without problems.
Thank you for your support.
